File: cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float f;
int n; //amount in cents
int coins;
int i; //counter

int main(void)
{
    do

    // get a float from the user
    {
    f = get_float("Amount paid:");
    }

    while(f <= 0);

coins = 0;


// convert f to an integer

n = round(f * 100);
// count out coins
if (n >= 1)
{
        for (i=0; n >= 25; coins++)
        {
            n = (n - 25);
        }
        
        for (i=0; n >= 10; coins++)
        {
            n = (n - 10);
        }
    
        for (i=0; n >= 5; coins++)
        {
            n = (n - 5);
        }
        for (i=0; n >= 1; coins++)
        {
            n = (n - 1);
        }
}
// print out the number of coins to return as change
printf("%i\n", coins);

}

File: pset6/sentimental-cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change owed: ")
    if change > -1:
        break

cents = int(change*100)
quarters = dimes = nickels = pennies = 0

# calculate quarters
if cents >= 25:
    quarters = cents // 25
    #print(f"q: {quarters}")
    cents = cents % 25

# calculate dimes
if cents >= 10:
    dimes = cents // 10
    #print(f"d: {dimes}")
    cents = cents % 10

# calculate nickels
if cents >= 5:
    nickels = cents // 5
    #print(f"n: {nickels}")
    cents = cents % 5

# calculate pennies
pennies = cents
#print(f"p: {pennies}")

coins = [quarters, dimes, nickels, pennies]
print(sum(coins))